feature/CATC_27-allow-users-to-add-a-list-to-a-board

sabellius opened the pull request:
## Description
This PR add functionality to allow users to create a new list directly from the board.

## Changes
- `UPDATE_BOARD` action also updates the board property in the store so that components that subscribe to updates of the board could re-render.
- `BoardDetails` not gets board state directly from the store instead of props. The page structure is wrong and needs to be fixed in a separate issue (`BoardIndex` renders `BoardDetails`).
- Add a simple AddForm component that uses local state to keep track of the list name, and on submit, updates the board lists array.

## Code review changes:
- Load boards using store actions in the `BoardIndex` compnent, and immediately navigate to the first boards details page.
-  Use `usParams` and load the board in `BoardDetails` from the store using the id from params.
- In the board-reducer, avoid updating the boards array in the `UPDATE_BOARD` actions, and instead, update the board object.

## Demo
[Screencast from 10-19-2025 06:56:59 PM.webm](https://github.com/user-attachments/assets/e1a88145-2ec9-4b4c-bdc0-c7920a9e48b8)

sabellius committed:
feat: use board from the global state instead of the component state

sabellius committed:
refactor: move styles to correct css file

sabellius committed:
feat: add a function for creating an empty list

sabellius committed:
fix: update current board state to reflect changes in the ui

sabellius committed:
feat: allow user to add a new list from the board

sabellius committed:
fix: get board state from the store instead of props

snoxlax commented on src/pages/BoardDetails.jsx:
```diff
@@ -1,20 +1,29 @@
-  StarBorderRounded,
-  LockOutlineRounded,
-} from "@mui/icons-material";
+export function BoardDetails() {
```
So I think this one should be with useParams() cuase if you remember we are doing 
`<Route path="board/:boardId" element={<BoardDetails />} />`
but I dont mind how it is right now if we just need it to work

sabellius replied:
Done

snoxlax commented on src/store/reducers/board-reducer.js:
```diff
@@ -28,7 +28,16 @@ export function boardReducer(state = initialState, action) {
         board._id === action.payload._id ? action.payload : board
       );
-      return { ...state, boards: updatedBoards };
+      const updatedCurrentBoard =
```
Shouldnt we put this in actions? it looks a bit complex

sabellius replied:
I think I should just update the board property here and not touch the board array.

snoxlax commented on src/pages/BoardIndex.jsx:
```diff
@@ -1,24 +1,21 @@
   }, []);
 
   async function fetchBoard() {
     const boards = await boardService.query();
```
Also here we should not use service I think we should get the boards from redux no?

LiorRey replied:
We should use loadBoards, or initially loadBoard(boardId) here, rather than using boardService in the Index.

sabellius replied:
I now get the board using the `params.boardId` and load it to the store and use it from there.

snoxlax reviewed (changes requested):
Other than what I wrote looks good!

sabellius committed:
fix: temporarily navigate from the index to the first board

This is used to to keep the index page functional while allowing to work
with a "default" board. This is a temporary solution until the full board
selection UI is implemented.

sabellius committed:
fix: load board using params and save it in the store

sabellius committed:
fix: avoid updating the boards array and instead only update the current board

snoxlax commented on src/components/AddList.jsx:
```diff
@@ -0,0 +1,54 @@
+  return (
+    <>
+      {showAddList && (
+        <Button
```
The styling looks off because its using an MUI Button.

Like I mentioned before, we might not need to rely on some of these components we could build our own custom ones ourselves.

That would make the design less dependent on the library among other things.

sabellius replied:
I mostly agree, there are some benefits using MUI components as they handle various things behind the scenes, but I also not too familiar with how easy is (or isn't) is to customize them. Regardless as we discussed in Discord, I'm keeping it ugly as it is now to focus more on functionality.

LiorRey commented on src/services/board/board-service-local.js:
```diff
@@ -78,6 +79,14 @@ async function updateBoardWithActivity(
   }
 }
 
+export function getEmptyList() {
```
Shouldn't it be in the `services/board/index.js`?

snoxlax reviewed (approved):
Looks great!

LiorRey reviewed (approved):
After we've discussed and settled things on Discord - I approve 🦾
